src/ppx/command_line_parser.cpp

```diff
@@ -37,157 +37,124 @@ bool IsOptionOrFlag(const std::string& s)
 
 namespace ppx {
 
+std::pair<int, int> CliOptions::GetOptionValueOrDefault(std::string_view optionName, const std::pair<int, int>& defaultValue) const
+{
+    auto it = allOptions.find(optionName);
+    if (it == allOptions.cend()) {
+        return defaultValue;
+    }
+    auto valueStr = it->second.back();
+    auto res      = ppx::string_util::SplitInTwo(valueStr, 'x');
+    if (res == std::nullopt) {
+        PPX_LOG_ERROR("resolution flag must be in format <Width>x<Height>: " << valueStr);
+        return defaultValue;
+    }
+    int N = GetParsedOrDefault(res->first, defaultValue.first);
+    int M = GetParsedOrDefault(res->second, defaultValue.second);
+    return std::make_pair(N, M);
+}
+
+void CliOptions::AddOption(std::string_view key, std::string_view valueStr)
```

Comment:
How about renaming key to something more descriptive to improve readability? Like optionName.